src/uct/ib/rc/accel/rc_mlx5_ep.c

```diff
@@ -431,8 +562,81 @@ ucs_status_t uct_rc_mlx5_ep_fc_ctrl(uct_ep_t *tl_ep, unsigned op,
 }
 
 #if IBV_EXP_HW_TM
-ucs_status_t uct_rc_mlx5_ep_tag_eager_short(uct_ep_h tl_ep, uct_tag_t tag,
-                                            const void *data, size_t length)
+
+#if HAVE_IBV_EXP_DM
+typedef struct uct_rc_mlx5_tm_dm_pack_data {
+    size_t     length;
+    const void *payload;
+} uct_rc_mlx5_tm_dm_pack_data_t;
+
+static size_t uct_rc_mlx5_ep_tm_short_dm_pack(void *dest, void *arg)
+{
+    uct_rc_mlx5_tm_dm_pack_data_t *data = (uct_rc_mlx5_tm_dm_pack_data_t*)arg;
+    memcpy(dest, data->payload, data->length);
+    return data->length;
+}
+
+ucs_status_t
+uct_rc_mlx5_ep_tag_eager_short_dm(uct_ep_h tl_ep, uct_tag_t tag,
+                                  const void *data, size_t length)
+{
+    uct_rc_mlx5_ep_t *ep  = ucs_derived_of(tl_ep, uct_rc_mlx5_ep_t);
+    uct_rc_iface_t *iface = ucs_derived_of(tl_ep->iface, uct_rc_iface_t);
+    uct_rc_iface_send_desc_t *desc;
+    uct_rc_mlx5_tm_dm_pack_data_t pack_data;
+    uct_rc_mlx5_dm_copy_data_t cache;
+    uint32_t app_ctx, ib_imm;
+    int opcode;
+    size_t len;
+
+    if (length + sizeof(struct ibv_exp_tmh) < UCT_IB_MLX5_AM_MAX_SHORT(0)) {
+        return uct_rc_mlx5_ep_tag_eager_short_inline(tl_ep, tag, data, length);
+    } else {
+        UCT_CHECK_LENGTH(length + sizeof(struct ibv_exp_tmh), 0,
+                         iface->dm.seg_len, "tag_short");
+        UCT_RC_CHECK_RES(iface, &ep->super);
+        UCT_RC_IFACE_FILL_TM_IMM(0, app_ctx, ib_imm, opcode, MLX5_OPCODE_SEND,
```

Comment:
you may avoid this here. Just use 0 instead of app_ctx and ib_imm and MLX5_OPCODE_SEND as an opcode

Reply:
used standard macro for cases if it could be changed later (compiler optimize it)

Reply:
You may avoid using `app_ctx`,  `ib_imm` and `opcode` at all, this would look simpler. Otherwise it looks a bit confusing now

Reply:
simpler than 1-line code?
for now it is standard way to prepare data for bcopy, in case if ```UCT_RC_IFACE_FILL_TM_IMM``` will be changed - such changes automatically will be reflected to DM path

Reply:
yes, because there will be no this line and several vars at all. Would be better to avoid unnecessary code in PRM transport

Reply:
ok, updated